# **Logistic Regression for Loan Approval**

In [2]:
import pandas as pd

In [4]:
LoanData = pd.read_csv('Loan Prediction Data.csv')
LoanData.head()

gender married   ch  income  loanamt status
0   Male      No  1.0    5849      NaN      Y
1   Male     Yes  1.0    4583    128.0      N
2   Male     Yes  1.0    3000     66.0      Y
3   Male     Yes  1.0    2583    120.0      Y
4   Male      No  1.0    6000    141.0      Y

In [6]:
LoanPrep = LoanData.copy()

# Remove missing values from the list

In [9]:
LoanPrep.isnull().sum(axis=0)

gender     13
married     3
ch         50
income      0
loanamt    22
status      0
dtype: int64

In [11]:
LoanPrep= LoanPrep.dropna()

In [13]:
LoanPrep.isnull().sum(axis=0)

gender     0
married    0
ch         0
income     0
loanamt    0
status     0
dtype: int64

# Drop unnecessary variables

<font size="4px"><p>In this case, gender does not play crucial role in getting loan approved. Thus, gender variable is removed</p>

In [18]:
LoanPrep = LoanPrep.drop(['gender'],axis=1)

In [20]:
LoanPrep.dtypes

married     object
ch         float64
income       int64
loanamt    float64
status      object
dtype: object

# Make dummy variables

In [23]:
LoanPrep = pd.get_dummies(LoanPrep, drop_first=True)

# Normalize

In [26]:
from sklearn.preprocessing import StandardScaler
scaler_= StandardScaler()

In [28]:
LoanPrep['income'] = scaler_.fit_transform(LoanPrep[['income']])
LoanPrep['loanamt'] = scaler_.fit_transform(LoanPrep[['loanamt']])

In [30]:
Y= LoanPrep[['status_Y']]
X= LoanPrep.drop(['status_Y'], axis=1)

In [32]:
from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test=\
    train_test_split(X, Y, test_size = 0.3, random_state = 1234, stratify=Y)

# Regression

In [36]:
from sklearn.linear_model import LogisticRegression
lr=LogisticRegression()

In [38]:
lr.fit(X_train, Y_train)

C:\Users\Shyl\anaconda3\Lib\site-packages\sklearn\utils\validation.py:1339: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


LogisticRegression()

In [40]:
Y_predict = lr.predict(X_test)

# Confusion Matrix
#True Negatives, True Positives, False Negatives, False Positives

In [47]:
from sklearn.metrics import confusion_matrix
cm= confusion_matrix(Y_test, Y_predict)
print(cm)

[[ 29  20]
 [  2 108]]


In [49]:
score = lr.score(X_test, Y_test)
print(score)

0.8616352201257862


# Conclusion

<font size="4px"><p>True negatives: 29, True positives:  108, False positives: 20, False negatives: 2</p>
<font size="4px"><p>The Accuracy is 86%, only 14% of false perdiction</p>